src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,38 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  health = 100;
+  constructor(name) {
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+  // eslint-disable-next-line no-useless-constructor
+  constructor(name) {
+    super(name);
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  // eslint-disable-next-line no-useless-constructor
+  constructor(name) {
+    super(name);
+  }
+  bite(obj) {
+    if (obj.hidden === false) {
+      obj.health -= 50;
+    }
+    Animal.alive = Animal.alive.filter((aliveObj) => aliveObj.health > 0);
```

Comment:
Filtering the `Animal.alive` array in the `bite` method might not be the best approach if you want to keep track of all animal instances, including those with zero or negative health. Consider whether you want to maintain a separate list for 'dead' animals or handle this logic differently.